src/mesido/esdl/asset_to_component_base.py

```diff
@@ -323,6 +327,84 @@ def _pipe_get_diameter_and_insulation(self, asset: Asset) -> Tuple[float, list,
 
         return inner_diameter, insulation_thicknesses, conductivies_insulation
 
+    def _gas_pipe_get_diameter_and_roughness(self, asset: Asset) -> Tuple[float, float]:
+        """
+        There are multiple ways to specify pipe properties like inner-diameter and
+        pipe material and thickness.  The user specified nominal diameter (DN size)
+        takes precedence over potential user specified innerDiameter and material (while logging
+        warnings when either of these two variables are specified in combination with the pipe DN)
+        Parameters
+        ----------
+        asset : Asset pipe object with it's properties from ESDL
+
+        Returns
+        -------
+        pipe inner diameter, thickness and conductivity of each insulation layer
+        """
+
+        full_name = f"{asset.asset_type} '{asset.name}'"
+        if asset.attributes["innerDiameter"] and asset.attributes["diameter"].value > 0:
+            logger.warning(
+                f"{full_name}' has both 'innerDiameter' and 'diameter' specified. "
+                f"Diameter of {asset.attributes['diameter'].name} will be used."
+            )
+        if asset.attributes["material"] and asset.attributes["diameter"].value > 0:
+            logger.warning(
+                f"{full_name}' has both 'material' and 'diameter' specified. "
+                f"Insulation properties of {asset.attributes['diameter'].name} will be used."
+            )
+        if asset.attributes["diameter"].value == 0 and not asset.attributes["innerDiameter"]:
+            logger.error(f"{full_name}' has no DN size or innerDiameter specified. ")
+
+        edr_dn_size = None
+        if asset.attributes["diameter"].value > 0:
+            edr_dn_size = str(asset.attributes["diameter"].name)
+        elif not asset.attributes["innerDiameter"]:
+            logger.warning(f"{full_name}' has no DN size or innerDiameter specified. ")
+            edr_dn_size = "DN200"  # ideally I don't want us to set a default diameter size, we
+            # can get in this line in case innerDiameter is set to 0.0.
+
+        # NaN means the default values will be used
+        wall_roughness = math.nan
+
+        if edr_dn_size:
+            # Get insulation and diameter properties from EDR asset with this size.
+            edr_asset = self._gas_pipes[edr_dn_size]
+            inner_diameter = float(edr_asset["inner_diameter"])
+            wall_roughness = float(edr_asset["roughness"])
+        else:
+            assert asset.attributes["innerDiameter"]
```

Comment:
Updated in case it is provided in the esdl, otherwise a default should be used. 
Currently the wall_roughness of individual pipes is still set by a generic setting in the energy_system_options. To be updated with this wall_roughness later.
The default should then be the energy_system options

Reply:
But when I look at this function the name states it will return pipe diameter as well as wall roughness. So I expect it to always return these values, but this is not the case (for instance when asset.attributes["roughness"] =< 0.0 or if not specified, this is a possibility). There is also nothing mentioned in the docstring that under certain circumstances no wall roughness (actually NAN) will be returned. In fact nothing is stated about the wall roughness in the docstring.

Reply:
I've updated the docstring, and added the default value here, to have clarity.